### PR TITLE
[WIP]ROC-5122: Mediation CSV

### PR DESCRIPTION
**Pull request by KiranVarma02**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ROC-5122

### Change description ###
Generate Mediation CSV

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[* ] No
```

### CODE REVIEW

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/MediationRequest.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package uk.gov.hmcts.cmc.domain.models;
+
+import org.springframework.format.annotation.DateTimeFormat;
+
+import javax.validation.constraints.NotEmpty;
+import java.time.LocalDate;
+
+public class MediationRequest {
```
Comment (andrewwa-kainos) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/MediationRequest.java]: Could use lombok here too.
`@AllArgsConstructor` on the class, and `@Getter` on both fields

Comment (andrewwa-kainos) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/MediationRequest.java]: Could use lombok here too.
@AllArgsConstructor on the class, and @Getter on both fields

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/MediationRow.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+@Builder
+public class MediationRow {
+    @Builder.Default
+    private String siteId = "4";
+    @Builder.Default
+    private String caseType = "1";
+    @Builder.Default
+    private String checkList = "4";
+    @Builder.Default
+    private String partyStatus = "5";
```
Comment (andrewwa-kainos) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/MediationRow.java]: If these won't be changed, make them `final`?

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/MediationRow.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+    private String caseNumber;
+    private String amount;
+    private String partyType;
+    private String contactName;
+    private String contactNumber;
+
+
+    public List<String> toArray() {
```
Comment (andrewwa-kainos) [inline domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/MediationRow.java]: Better keep the return type and method name consistent; arrays and lists aren't the same in Java. 
I.e., `String[] toArray()` or `List<String> toList()`

File: src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/ClaimRepository.java [hunk 4]
```diff
@@ -68,6 +68,12 @@
     @SqlQuery(SELECT_FROM_STATEMENT + " WHERE claim.is_migrated = false")
     List<Claim> getAllNotMigratedClaims();
 
+    @SqlQuery(
+        SELECT_FROM_STATEMENT + " WHERE claim.response ->>'freeMediation = 'yes' " +
+            "AND claim.claimant_response ->>'freeMediation = 'yes'" +
+            "AND claim.claimant_responded_at = :claimant_responded_date")
+    List<Claim> getMediationClaimsForDate(@Bind("claimant_responded_date") LocalDate claimant_responded_date);
```
Comment (andrewwa-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/ClaimRepository.java]: Can give the java variable a `camelCase` name; the `@Bind` value is what needs to match the `@SqlQuery` placeholder.

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/MediationCSVService.java [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+        if(mediationClaims.isEmpty()) {
+            result.add(MediationRow.builder()
+                .siteId(null)
+                .caseType(null)
+                .checkList(null)
+                .partyStatus(null)
+                .build().toArray()
+            );
+        }
```
Comment (andrewwa-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/MediationCSVService.java]: Why can't we return an empty list?

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/MediationCSVService.java [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    private MediationRow createMediationRow (Claim claim, int partyType) {
+        Response response = claim.getResponse().orElseThrow(IllegalArgumentException::new);
+        return MediationRow.builder()
+            .caseNumber(claim.getReferenceNumber())
+            .amount(String.valueOf(claim.getTotalAmountTillToday().orElse(BigDecimal.ZERO)))
+            .partyType(String.valueOf(partyType))
+            .contactName(response.getMediationContactPerson().orElse(Strings.EMPTY))
+            .contactNumber(response.getMediationPhoneNumber().orElse(Strings.EMPTY))
```
Comment (andrewwa-kainos) [inline src/main/java/uk/gov/hmcts/cmc/claimstore/services/MediationCSVService.java]: `Strings.EMPTY` is coming from a logging library not really appropriate here. I'd rather see just `""`, personally, but doesn't matter much.

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/MediationCSVServiceTest.java [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+public class MediationCSVServiceTest {
+
+    private static final String AUTHORISATION = "Bearer: aaa";
+
+    private MediationCSVService mediationCSVService;
+
+    @Mock
+    private CaseRepository mockCaseRepository;
+
+    private List<Claim> mediationClaims = new ArrayList<>();
```
Comment (andrewwa-kainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/services/MediationCSVServiceTest.java]: Instantiate in `setUp()`

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/MediationCSVServiceTest.java [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+    @Before
+    public void setUp () {
```
Comment (andrewwa-kainos) [inline src/test/java/uk/gov/hmcts/cmc/claimstore/services/MediationCSVServiceTest.java]: Our Java style does not separate method names from parameter lists.

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/MediationRequest.java (hunk 1 above)
Comment (KiranVarma02) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/MediationRequest.java]: done

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/MediationRow.java (hunk 2 above)
Comment (KiranVarma02) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/MediationRow.java]: done

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/MediationRow.java (hunk 3 above)
Comment (KiranVarma02) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/MediationRow.java]: done

File: src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/ClaimRepository.java (hunk 4 above)
Comment (KiranVarma02) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/repositories/ClaimRepository.java]: done

File: domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/MediationRow.java (hunk 2 above)
Comment (KiranVarma02) [reply domain-model/src/main/java/uk/gov/hmcts/cmc/domain/models/MediationRow.java]: changed logic slightly but using final now

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/MediationCSVService.java (hunk 5 above)
Comment (KiranVarma02) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/MediationCSVService.java]: changed logic slightly to not have to pass null

File: src/main/java/uk/gov/hmcts/cmc/claimstore/services/MediationCSVService.java (hunk 6 above)
Comment (KiranVarma02) [reply src/main/java/uk/gov/hmcts/cmc/claimstore/services/MediationCSVService.java]: handled differently now, please check

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/MediationCSVServiceTest.java (hunk 7 above)
Comment (KiranVarma02) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/MediationCSVServiceTest.java]: done

File: src/test/java/uk/gov/hmcts/cmc/claimstore/services/MediationCSVServiceTest.java (hunk 8 above)
Comment (KiranVarma02) [reply src/test/java/uk/gov/hmcts/cmc/claimstore/services/MediationCSVServiceTest.java]: I assume you are talking about space between `setUp ()`!!, its corrected now

**Comment by KiranVarma02** [comment]
delete to start fresh

**Pull request closed without merging**